redash/destinations/asana.py

```diff
@@ -0,0 +1,58 @@
+import logging
+import textwrap
+
+import requests
+
+from redash.destinations import BaseDestination, register
+
+
+class Asana(BaseDestination):
+    @classmethod
+    def configuration_schema(cls):
+        return {
+            "type": "object",
+            "properties": {
+                "pat": {"type": "string", "title": "Asana Personal Access Token"},
+                "project_id": {"type": "string", "title": "Asana Project ID"},
+            },
+            "secret": ["pat"],
+        }
+
+    @classmethod
+    def icon(cls):
+        return "fa-sticky-note-o"
+
+    def notify(self, alert, query, user, new_state, app, host, options):
+        # Documentation: https://developers.asana.com/docs/tasks
+        state = "TRIGGERED" if new_state == "triggered" else "RECOVERED"
+
+        notes = textwrap.dedent(
+            f"""
+        {alert.name} has {state}.
+
+        Query: {host}/queries/{query.id}
+        Alert: {host}/alerts/{alert.id}
+        """
+        ).strip()
+
+        data = {
+            "name": f"[Redash Alert] {state}: {alert.name}",
+            "notes": notes,
+            "projects": [options.get("project_id")],
+        }
+
+        try:
+            resp = requests.post(
+                "https://app.asana.com/api/1.0/tasks",
```

Comment:
Maybe this line should be brought up and be outside the class / a separate method in the class declaration, to make it easier to find the URL being posted in case the API changes?